compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -619,6 +619,14 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
             ast::ExprKind::TryBlock(_) => {
                 gate_feature_post!(&self, try_blocks, e.span, "`try` expression is experimental");
             }
+            ast::ExprKind::Yeet(_) => {
+                gate_feature_post!(
+                    &self,
+                    yeet_expr,
+                    e.span,
+                    "`do yeet` expression is experimental"
+                );
```

Comment:
Feature gating at AST validation time doesn't gate syntax, e.g.
```rust
#[cfg(FALSE)]
fn check() {
    /* this cringeworthy syntax */
}
```
will still work on stable.

For an example of parse-time gating see e.g. `inline_const`.

(Yes, some unstable syntax, like `try_blocks` is not gated at parse-time, but that's for backward compatibility, new feature should use the parse-time gating.)

Reply:
Oh, thanks @petrochenkov!  I'll go do that instead.

@rustbot author